cargo-quickinstall/src/lib.rs

```diff
@@ -94,10 +101,10 @@ pub fn get_latest_version(crate_name: &str) -> Result<String, InstallError> {
                 e
             }
         })?
-        .get_owned(&"versions")
-        .and_then(|value| value.get_owned(&0))
-        .and_then(|value| value.get_owned(&"num"))
-        .and_then(JsonValueExt::try_into_string)
+        .extract_from_value("versions")?
+        .extract_from_value(0)?
+        .extract_from_value("num")?
+        .try_into_string()
         .map_err(InstallError::from)
```

Comment:
(optional) if I'm not using Ok(err?) then I would normally write From::from (that what ? desugars into anyway)). This is one less thing to refactor if you change your error types later.

```suggestion
        .map_err(From::from)
```